src/views/ConfusionMatrix/ConfusionMatrixView.tsx

```diff
@@ -0,0 +1,118 @@
+import { INITIALIZE_UNITS, useSelectedUnitIds } from 'contexts/UnitSelection/UnitSelectionContext'
+import { FunctionComponent, useCallback, useEffect, useMemo, useState } from 'react'
+import BottomToolbar from 'views/UnitSimilarityMatrix/BottomToolbar'
+import { sortIds } from 'views/UnitsTable/UnitsTableView'
+import MatrixWidget from '../UnitSimilarityMatrix/MatrixWidget'
+import { ConfusionMatrixViewData } from './ConfusionMatrixViewData'
+
+type Props = {
+    data: ConfusionMatrixViewData
+    width: number
+    height: number
+}
+
+const defaultRange: [number, number] = [0, 1]
+
+export type HoveredInfo = {
+    unitId1: number | string
+    unitId2: number | string
+    value: number | undefined
+}
+
+const ConfusionMatrixView: FunctionComponent<Props> = ({ data, width, height }) => {
+    const { selectedUnitIds, unitIdSelectionDispatch } = useSelectedUnitIds()
+    const [hoveredInfo, setHoveredInfo] = useState<HoveredInfo | undefined>(undefined)
+
+    const {sorting1UnitIds, sorting2UnitIds, unitEventCounts, matchingUnitEventCounts} = data
+
+    const {allUnitIds1, allUnitIds2} = useMemo(() => (
+        {
+            allUnitIds1: sortIds([...sorting1UnitIds]),
+            allUnitIds2: sortIds([...sorting2UnitIds])
+        }
+    ), [sorting1UnitIds, sorting2UnitIds])
+
+    const allUnitIds = useMemo(() => ([...allUnitIds1, ...allUnitIds2]), [allUnitIds1, allUnitIds2])
+
+    useEffect(() => {
+        unitIdSelectionDispatch({ type: INITIALIZE_UNITS, newUnitOrder: allUnitIds})
+    }, [allUnitIds, unitIdSelectionDispatch])
+
+    const eventCountsForIds = useMemo(() => {
+        const ret: {[id: number | string]: number} = {}
+        for (let x of unitEventCounts) {
+            ret[x.unitId] = x.count
+        }
+        return ret
+    }, [unitEventCounts])
```

Comment:
Doesn't this potentially overwrite if the two sortings use the same naming scheme for their unit ids?

Reply:
the IDs are mutually exclusive